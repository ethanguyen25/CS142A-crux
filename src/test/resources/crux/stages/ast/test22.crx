// test using expression as index in assignment

int a[3];
int b[10];
  
void main() {

   b[5] = 2;
   a[1+1] = 100;

  printInt(a[2]);
}

