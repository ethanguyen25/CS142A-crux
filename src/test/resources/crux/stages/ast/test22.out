ast.DeclarationList(3)
  ast.ArrayDeclaration(3)[Symbol(a)]
  ast.ArrayDeclaration(4)[Symbol(b)]
  ast.FunctionDefinition(6)[Symbol(main), []]
    ast.StatementList(8)
      ast.Assignment(8)
        ast.ArrayAccess(8)
          ast.Name(8)[b]
          ast.LiteralInt(8)[5]
        ast.LiteralInt(8)[2]
      ast.Assignment(9)
        ast.ArrayAccess(9)
          ast.Name(9)[a]
          ast.OpExpr(9)[ADD]
            ast.LiteralInt(9)[1]
            ast.LiteralInt(9)[1]
        ast.LiteralInt(9)[100]
      ast.Call(11)[Symbol(printInt)]
        ast.Dereference(11)
          ast.ArrayAccess(11)
            ast.Name(11)[a]
            ast.LiteralInt(11)[2]
